docs/how-to/troubleshoot_deployment_issues.md

```diff
@@ -0,0 +1,115 @@
+│ model.juju.is/id:9755936b-8084-4397-8a67-28773b361dfa] (not valid)
+```
+
+### Recommended Actions
+
+Validate whether a Juju model already exists with the same name:
+
+```shell
+$ juju models
+controller: microk8s-localhost
+
+Model       Cloud/Region        Type        Status      Units  Access  Last connection
+controller  microk8s/localhost  kubernetes  available   1       admin  just now
+private5g*  microk8s/localhost  kubernetes  destroying  -       admin  6 minutes ago
+sdcore      microk8s/localhost  kubernetes  available   19      admin  2024-10-04
+```
+
+Choose a model name that does not exist in the Juju controller. Read [this guide][Configure SD-Core K8s Deployment] for more information.
+
+## 2. Terraform failed to deploy Charmed Aether SD-Core with `Connection error, please check the controller_addresses property set on the provider` error
+
+### Symptoms
+
+The `terraform apply -auto-approve` command fails with an error indicating that the Juju controller couldn't be connected to:
+
+```console
+$ terraform apply --auto-approve
+│ Error: Invalid provider configuration
+│ Provider "registry.terraform.io/juju/juju" requires explicit configuration. Add a provider block to the root module and configure the provider's required arguments as described in
+│ the provider documentation.
+│ Error: dial tcp 10.152.183.251:17070: i/o timeout
+│   with provider["registry.terraform.io/juju/juju"],
+│   on <empty> line 0:
+│   (source code not available)
+│ Connection error, please check the controller_addresses property set on the provider
+```
+
+### Recommended Actions
+
+Validate that the Juju controller is running.
+
+First, list the available Juju controllers:
+
+```shell
+$ juju controllers
+Use --refresh option with this command to see the latest information.
+
+Controller           Model      User   Access     Cloud/Region        Models  Nodes  HA  Version
+microk8s-localhost*  private5g  admin  superuser  microk8s/localhost       9      -   -  3.4.5  
+```
+
+If your controller does not show up in the list, please follow [this guide][Bootstrap a Juju Controller] to create a Juju controller.
+
+If the controller is listed, get your controller's `api-endpoints` address.
+
+```shell
+$ juju show-controller <your-controller-name>
+microk8s-localhost:
+  details:
+    controller-uuid: ced7016b-3a63-4133-8988-cf33068c3cdf
+    api-endpoints: ['10.152.183.251:17070']
+    cloud: microk8s
+    region: localhost
+    agent-version: 3.4.5
+```
+
+Perform a healthcheck using your Juju controller's `api-endpoints` address which is `10.152.183.251:17070` in this guide:
+
+```shell
+$ curl -ik https://<api-endpoints>/health
+HTTP/1.1 200 OK
+Date: Thu, 10 Oct 2024 12:39:00 GMT
+Content-Length: 8
+Content-Type: text/plain; charset=utf-8
+
+running
+```
+
+If the healthcheck returns `running` check the firewall rules in your environment. Otherwise, access to the controller `api-server` container and check the logs:
```

Comment:
```suggestion
If the healthcheck returns `running` check the firewall rules in your environment. Otherwise, check the controller's `api-server` container logs for any errors:
```

Reply:
This is removed.